vaadin-testbench-core/src/main/kotlin/com/vaadin/testbench/unit/internal/ComponentUtils.kt

```diff
@@ -0,0 +1,393 @@
+import com.vaadin.flow.component.ClickNotifier
+import com.vaadin.flow.component.Component
+import com.vaadin.flow.component.ComponentEvent
+import com.vaadin.flow.component.ComponentUtil
+import com.vaadin.flow.component.HasComponents
+import com.vaadin.flow.component.HasElement
+import com.vaadin.flow.component.HasOrderedComponents
+import com.vaadin.flow.component.HasStyle
+import com.vaadin.flow.component.HasText
+import com.vaadin.flow.component.UI
+import com.vaadin.flow.component.button.Button
+import com.vaadin.flow.component.checkbox.Checkbox
+import com.vaadin.flow.component.checkbox.CheckboxGroup
+import com.vaadin.flow.component.combobox.ComboBox
+import com.vaadin.flow.component.datepicker.DatePicker
+import com.vaadin.flow.component.formlayout.FormLayout
+import com.vaadin.flow.component.grid.Grid
+import com.vaadin.flow.component.html.Input
+import com.vaadin.flow.component.ironlist.IronList
+import com.vaadin.flow.component.listbox.ListBoxBase
+import com.vaadin.flow.component.radiobutton.RadioButtonGroup
+import com.vaadin.flow.component.select.Select
+import com.vaadin.flow.component.textfield.PasswordField
+import com.vaadin.flow.component.textfield.TextArea
+import com.vaadin.flow.component.textfield.TextField
+import com.vaadin.flow.data.binder.HasDataProvider
+import com.vaadin.flow.data.provider.DataProvider
+import com.vaadin.flow.dom.DomEventListener
+import com.vaadin.flow.dom.DomListenerRegistration
+import com.vaadin.flow.router.Location
+
+/**
+ * Fires given event on the component.
+ */
+fun Component.fireEvent(event: ComponentEvent<*>) {
+    ComponentUtil.fireEvent(this, event)
+}
+
+/**
+ * Adds [com.vaadin.flow.component.button.Button.click] functionality to all [ClickNotifier]s. This function directly calls
+ * all click listeners, thus it avoids the roundtrip to client and back. It even works with browserless testing.
+ * @param fromClient see [ComponentEvent.isFromClient], defaults to true.
+ * @param button see [ClickEvent.getButton], defaults to 0.
+ * @param clickCount see [ClickEvent.getClickCount], defaults to 1.
+ */
+fun <T: ClickNotifier<*>> T.serverClick(
+    fromClient: Boolean = true,
+    button: Int = 0,
+    clickCount: Int = 1,
+    shiftKey: Boolean = false,
+    ctrlKey: Boolean = false,
+    altKey: Boolean = false,
+    metaKey: Boolean = false
+) {
+    (this as Component).fireEvent(ClickEvent<Component>(this,
+        fromClient, -1, -1, -1, -1, clickCount, button, ctrlKey, shiftKey, altKey, metaKey))
+}
+
+/**
+ * Sets the alignment of the text in the component. One of `center`, `left`, `right`, `justify`.
+ */
+var Component.textAlign: String?
+    get() = element.style.get("textAlign")
+    set(value) { element.style.set("textAlign", value) }
+
+/**
+ * Sets or removes the `title` attribute on component's element.
+ */
+var Component.tooltip: String?
+    get() = element.getAttribute("title")
+    set(value) { element.setOrRemoveAttribute("title", value) }
```

Comment:
This sounds a bit strange as an addition for testing as the feature doesn't exist.

Reply:
perhaps drop any extension that adds get set for attributes.